xdvdfs-core/src/write/img.rs

```diff
@@ -148,7 +149,7 @@ pub async fn create_xdvdfs_image<H: BlockDeviceWrite<E>, E>(
         .await?;
 
         for entry in dirtab.file_listing {
-            let file_path = path.join(&entry.name);
+            let file_path = path.join_xdvdfs(&entry.name);
```

Comment:
I'm thinking of three possible ways of doing this:

1. The way you've suggested, by changing the signature of `copy_file_in` and `copy_file_buf`, allowing to join the path there.
2. By replacing the `\` character with `/` at the `copy_file_in` method from `XDVDFSFilesystem` .
3. Using conditional compilation to call `join_xdvdfs` here.

The first option would probably be the best way around this, but the FS module would lose some of its generic-ness, which seems to go against your initial design idea.

The second one would take advantage of the fact that filenames shouldn't contain `/` or `\` characters within XDVDFS archives, as is with FATX volumes. But I'm not sure `xdvdfs` as a project is enforcing this rule, as Microsoft tools did.

And the last one is mostly a hacky way around it, by having Windows targets use the right thing at the right time.

Sorry for the long comment. Please let me know what you think. @antangelo

Reply:
I had option 2 in mind. I don't believe `/` or `\` are valid characters in the filename so that tradeoff is fine (and if they were valid, the read functions wouldn't be able to handle them as they are now anyway).